### PR TITLE
Use a browser auth flow to support Google login

### PR DESCRIPTION
**Pull request by mbleigh**
Update Vulcan to use the same login flow as the CLI, allowing it to work properly with Google login accounts.

### CODE REVIEW

**Comment by mbleigh** [comment]
@tstirrat looks like maybe you're the person to review this...my first time playing around with Vulcan code so please let me know if anything looks off.

**Comment by tstirrat** [comment]
@joeyyang this looks good to me, leaving final review to you

File: app/js/components/form-login.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@
 var React = require('react/addons');
 var AppMixins = require('./mixins');
 var $ = require('jquery');
+var Firebase = require('firebase');
```
Comment (tstirrat) [inline app/js/components/form-login.jsx]: looks like `Firebase` is not used in the file

Comment (mbleigh) [inline app/js/components/form-login.jsx]: Ah, yes, I originally implemented the ticket in here but realized that was the wrong place.

File: app/js/components/container.jsx [hunk 2]
```diff
@@ -196,50 +196,74 @@ module.exports = React.createClass({
 
 
   /**
+   * generateUUID
+   *
+   * Generate and return a random (v4) uuid
+   */
+  generateUUID: function() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+        var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
```
Comment (joeyyang) [inline app/js/components/container.jsx]: could you leave a comment here for future generations?

File: app/js/components/container.jsx [hunk 3]
```diff
@@ -196,50 +196,74 @@ module.exports = React.createClass({
...
+   */
+  ticketUpdated: function(snap) {
+    if (!snap.exists()) {
+      return;
+    }
+
+    var self = this;
+
+    var adminToken = snap.child('session/token').val();
+    self._setAdminToken(adminToken);
```
Comment (joeyyang) [inline app/js/components/container.jsx]: `this`

**Comment by mbleigh** [comment]
@joeyyang switched to the uuid-v4 library to generate the uuid.

**Comment by joeyyang** [comment]
The error state here is specific to the CLI. Could you also submit a companion PR to the website to handle the Vulcan case?
screen shot 2015-10-22 at 9 12 29 am

**Comment by joeyyang** [comment]
The sign-in flow is currently busted for me. Could you investigate? (Back to you.)

**Comment by mbleigh** [comment]
Sure. Would prefer that not gate merge though.

**Comment by joeyyang** [comment]
@mbleigh sorry, here's more error info — let me know if I can help repro: 
screen shot 2015-10-22 at 9 43 33 am

**Pull request merged**